BinaryObjectScanner/Protection/CopyX.cs

```diff
@@ -0,0 +1,182 @@
+                // TODO: Find a way to check for situations like Redump ID 48393, where the string is spaced out with
+                // 0x00 between letters and does not show up on string checks.
+                // TODO: This might need to check every single section. Unsure until more samples are acquired.
+                if (pex.OverlayStrings.Any(s => s.Contains("optgraph.dll")))
+                {
+                    // TODO: TKKG also has an NE 3.1x executable with a reference. This can be added later.
+                    // Samples: Redump ID 108150
+                    return "copy-X";
+                }
+            }
+
+            var strs = pex.GetFirstSectionStrings(".rdata");
+            if (strs != null)
+            {
+                // Samples: Redump ID 82475 (False positive, but probably catches original Emergency 2), would catch Redump ID 48393
+                if (strs.Any(s => s.Contains("optgraph.dll")))
+                    return "copy-X";
+            }
+
+            return null;
+        }
+
+        /// <inheritdoc/>
+#if NET20 || NET35
+        public Queue<string> CheckDirectoryPath(string path, IEnumerable<string>? files)
+#else
+        public ConcurrentQueue<string> CheckDirectoryPath(string path, IEnumerable<string>? files)
+#endif
+        {
+#if NET20 || NET35
+            var protections = new Queue<string>();
+#else
+            var protections = new ConcurrentQueue<string>();
+#endif
+
+            // Checks for Light
+            if (files == null)
+                return protections;
+
+            // Gets files in ZDAT*
+            // Excludes files with .x64 extension to avoid flagging Professional files.
+            var fileList = files.Where(f => !f.EndsWith(".x64", StringComparison.OrdinalIgnoreCase)).Where(f =>
+            {
+                f = f.Remove(0, path.Length);
+                f = f.TrimStart('/', '\\');
+                return f.StartsWith("ZDAT", StringComparison.OrdinalIgnoreCase);
+            }).OrderBy(f => f).ToList();
+            // Sorts list of files in ZDAT* so just the first file gets pulled, later ones have a chance of the ring
+            // intersecting the start of the file.
+            if (fileList.Count > 0)
+            {
+                try
+                {
+                    using var stream = File.OpenRead(fileList[0]);
+                    byte[] block = stream.ReadBytes(1024);
+
+                    var matchers = new List<ContentMatchSet>
+                    {
+                        // Checks if the file contains 0x00
+                        // Samples: Redump ID 81628
+                        new(Enumerable.Repeat<byte?>(0x00, 1024).ToArray(), "copy-X"),
+                        // Checks for whatever this data is.
+                        // Samples: Redump ID 84759, Redump ID 107929. Professional discs also have this data, hence the exclusion check.
+                        new(new byte?[]
```

Comment:
Easy to miss, but here's a slightly nicer syntax you can use. Instead of:
```c#
new byte?[] { 0x01, 0x02, 0x03, 0x04 }
```
you can use
```c#
[0x01, 0x02, 0x03, 0x04]
```
Since it already knows what the type of the array is.